File: gobii-api-model/src/main/java/org/gobiiproject/gobiiapimodel/payload/Pagination.java
package org.gobiiproject.gobiiapimodel.payload;

import java.util.Date;

/**
 * Created by Phil on 9/6/2016.
 */
public class Pagination {


    public Pagination() {}

    public Pagination(String pagedQueryId,
                      Date queryTime,
                      Integer pageSize,
                      Integer totalPages,
                      Integer currentPage
    ) {
        this.pagedQueryId = pagedQueryId;
        this.queryTime = queryTime;
        this.pageSize= pageSize;
        this.totalPages= totalPages;
        this.currentPage= currentPage;

    }

    private String pagedQueryId;
    private Date queryTime;
    private Integer pageSize;
    private Integer totalPages;
    private Integer currentPage;


    public String getPagedQueryId() {
        return pagedQueryId;
    }

    public Date getQueryTime() {
        return queryTime;
    }

    public Integer getPageSize() {
        return pageSize;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public Integer getCurrentPage() {
        return currentPage;
    }
}

File: gobii-web/src/extractor-ui/src/model/payload/pagination.ts


export class Pagination {

    public constructor(public currentPage: number,
                       public pageSize: number,
                       public pagedQueryId: string,
                       public queryTime: string,
                       public totalPages: number) {
    }

    public static fromJSON(json: any): Pagination {


        let currentPage: number = json.currentPage;
        let pageSize: number = json.pageSize;
        let pagedQueryId: string = json.pagedQueryId;
        let queryTime: string = json.queryTime;
        let totalPages: number = json.totalPages;


        return new Pagination(
            currentPage,
            pageSize,
            pagedQueryId,
            queryTime,
            totalPages
        ); // new

    } // fromJson()
}
